src/model/Series.ts

```diff
@@ -582,26 +601,42 @@ class SeriesModel<Opt extends SeriesOption = SeriesOption> extends ComponentMode
     }
 
     private _innerSelect(data: SeriesData, innerDataIndices: number[]) {
-        const selectedMode = this.option.selectedMode;
+        const option = this.option;
+        const selectedMode = option.selectedMode;
         const len = innerDataIndices.length;
         if (!selectedMode || !len) {
             return;
         }
 
-        if (selectedMode === 'multiple') {
-            const selectedMap = this.option.selectedMap || (this.option.selectedMap = {});
+        if (selectedMode === 'series') {
+            option.selectedMap = 'all';
+        }
+        else if (selectedMode === 'multiple') {
+            if (!isObject(option.selectedMap)) {
+                option.selectedMap = {};
+            }
+            const selectedMap = option.selectedMap;
             for (let i = 0; i < len; i++) {
                 const dataIndex = innerDataIndices[i];
+                if ((data.getItemModel(dataIndex) as Model).get(['select', 'disabled']) === true) {
+                    return;
+                }
                 // TODO diffrent types of data share same object.
                 const nameOrId = getSelectionKey(data, dataIndex);
                 selectedMap[nameOrId] = true;
                 this._selectedDataIndicesMap[nameOrId] = data.getRawIndex(dataIndex);
             }
         }
         else if (selectedMode === 'single' || selectedMode === true) {
+            if (isString(option.selectedMap)) {
+                option.selectedMap = {};
+            }
             const lastDataIndex = innerDataIndices[len - 1];
+            if ((data.getItemModel(lastDataIndex) as Model).get(['select', 'disabled']) === true) {
```

Comment:
Current `as Model` lacks the type check. Perhaps we can remove this `as` and specify a generic type for `getItemModel`.
```ts
data.getItemModel<StatesOptionMixin<unknown, unknown>>(dataIndex).get(['select', 'disabled'])
```

Besides, I personally think we can remove `=== true`, that is, it allows the **truthy** value, e.g, `1`. And this also reduces the code slightly.